docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+
+
+###  Module 3 BFD state pinned
+Here the BFD pin down refers to a upstream service provided state, which does not essentially indicate a link failure hence does not trigger an actual switchover.
+
+| Case                                    | Goal                                                | Test Steps                                                                                     | Expected Control Plane Behavior                               | Expected Data Plane Behavior                      |
+|-----------------------------------------|-----------------------------------------------------|------------------------------------------------------------------------------------------------|--------------------------------------------------------------|---------------------------------------------------|
+| BFD state UP pinned as DOWN – Active    | Verify traffic flow honored the pinned state.       | • Start  I/O through active side<br>• Pin DPU1 BFD probe state as DOWN.             | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption.          |
+| BFD state UP pinned as DOWN – Standby   | Verify traffic flow honored the pinned state.       | • Start  I/O through standby side<br>• Pin DPU1 BFD probe state as DOWN.            | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption.          |
+| Both side pinned as DOWN – Active       | Verify traffic flow honored the pinned state.       | • Start  I/O through active side<br>• Pin DPU1 and DPU2 BFD probe state as DOWN.   | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption.          |
+| Both side pinned as DOWN – Standby      | Verify traffic flow honored the pinned state.       | • Start  I/O through standby side<br>• Pin DPU1 and DPU2 BFD probe state as DOWN.  | DPU1 remains active, DPU2 remains standby.                   | T2 receives packets without disruption.          |
+
+
+###  Module 4 HA state pinned by upstream service
+
+| Case                           | Goal                                           | Test Steps                                                                                                            | Expected Control Plane Behavior                   | Expected Data Plane Behavior                      |
+|--------------------------------|------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|--------------------------------------------------|---------------------------------------------------|
+| HA state pinned – Active       | Verify control plane honors the pinned state. | • Start  I/O through DPU1.<br>• Pin DPU2 as standalone.<br>• Remove Pin.                                    | DPU1 becomes standby, DPU2 becomes active.        | T2 receives packets without disruption.          |
```

Comment:
we might need test to test the behavior of pinning to standalone

Reply:
Yes, both cases are pinned standalone. 

Confirming offline to see I understand the comment correctly.